presto-main/src/main/java/com/facebook/presto/metadata/SqlAggregationFunction.java

```diff
@@ -61,15 +63,17 @@ protected SqlAggregationFunction(
             List<LongVariableConstraint> longVariableConstraints,
             TypeSignature returnType,
             List<TypeSignature> argumentTypes,
-            FunctionKind kind)
+            FunctionKind kind,
+            boolean deprecated)
     {
-        this(createSignature(name, typeVariableConstraints, longVariableConstraints, returnType, argumentTypes, kind), false);
+        this(createSignature(name, typeVariableConstraints, longVariableConstraints, returnType, argumentTypes, kind), false, deprecated);
     }
 
-    protected SqlAggregationFunction(Signature signature, boolean hidden)
+    protected SqlAggregationFunction(Signature signature, boolean hidden, boolean deprecated)
```

Comment:
Similar to `hidden`, should we just put the `deprecated` flag here, and default to `false` for the others?  CC: @haozhun

Reply:
I think this will reduce the scope of this PR significantly while still preserving the desired opt-in deprecation behavior.